mesos_master/datadog_checks/mesos_master/mesos_master.py

```diff
@@ -135,21 +135,31 @@ class MesosMaster(AgentCheck):
         'master/valid_status_updates': ('mesos.cluster.valid_status_updates', GAUGE),
     }
 
-    def __init__(self, name, init_config, agentConfig, instances=None):
-        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
+    HTTP_CONFIG_REMAPPER = {'disable_ssl_validation': {'name': 'tls_verify', 'invert': True, 'default': False}}
+
+    def __init__(self, name, init_config, instances):
+        super(MesosMaster, self).__init__(name, init_config, instances)
         for instance in instances or []:
             url = instance.get('url', '')
             parsed_url = urlparse(url)
-            ssl_verify = not _is_affirmative(instance.get('disable_ssl_validation', False))
-            if not ssl_verify and parsed_url.scheme == 'https':
-                self.log.warning('Skipping SSL cert validation for %s based on configuration.' % url)
-
-    def _get_json(self, url, timeout, verify=True, tags=None):
+            if not self.http.options['verify'] and parsed_url.scheme == 'https':
+                self.log.warning('Skipping TLS cert validation for %s based on configuration.' % url)
+            if not ('read_timeout' in instance or 'connect_timeout' in instance):
+                # `default_timeout` config option will be removed with Agent 5
+                self.http.options['timeout'] = (
+                    instance.get('timeout')
+                    or self.init_config.get('timeout')
+                    or self.init_config.get('default_timeout')
+                    or self.DEFAULT_TIMEOUT
+                )
```

Comment:
Seems that this could have the same effect. What do you think ?

```python
    def __init__(self, name, init_config, instances):
        super(MesosMaster, self).__init__(name, init_config, instances)
        for instance in instances or []:
            url = instance.get('url', '')
            parsed_url = urlparse(url)
            if not self.http.options['verify'] and parsed_url.scheme == 'https':
                self.log.warning('Skipping TLS cert validation for %s based on configuration.' % url)

        self.HTTP_CONFIG_REMAPPER = {
            'disable_ssl_validation': {'name': 'tls_verify', 'invert': True, 'default': False},
            'timeout': {'name': 'timeout', 'default': self.init_config.get('default_timeout', DEFAULT_TIMEOUT)},
        }

```

Reply:
Is there an advantage moving it into the `__init__`? Some integrations don't have them so keeping them outside of `__init__` was consistent.

Reply:
👍, not a big deal, both should work equally fine.